zmon_worker_monitor/builtins/plugins/appdynamics.py

```diff
@@ -0,0 +1,164 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+import os
+import sys
+import logging
+
+import requests
+
+import tokens
+
+from zmon_worker_monitor.zmon_worker.errors import HttpError
+from zmon_worker_monitor.zmon_worker.common.http import get_user_agent
+
+from zmon_worker_monitor.adapters.ifunctionfactory_plugin import IFunctionFactoryPlugin, propartial
+
+logger = logging.getLogger('zmon-worker.appdynamics-function')
+
+
+BEFORE_NOW = 'BEFORE_NOW'
+BEFORE_TIME = 'BEFORE_TIME'
+AFTER_TIME = 'AFTER_TIME'
+BETWEEN_TIMES = 'BETWEEN_TIMES'
+
+CRITICAL = 'CRITICAL'
+WARNING = 'WARNING'
+
+TIME_RANGE_TYPES = (
+    BEFORE_NOW,
+    BEFORE_TIME,
+    AFTER_TIME,
+    BETWEEN_TIMES,
+)
+
+SEVERITIES = (
+    WARNING,
+    CRITICAL,
+)
+
+
+# will use OAUTH2_ACCESS_TOKEN_URL environment variable by default
+# will try to read application credentials from CREDENTIALS_DIR
+tokens.configure()
+tokens.manage('uid', ['uid'])
+tokens.start()
+
+
+class AppdynamicsFactory(IFunctionFactoryPlugin):
+    def __init__(self):
+        super(AppdynamicsFactory, self).__init__()
+        # Fields should be fetched from config
+        self._user = None
+        self._pass = None
+
+    def configure(self, conf):
+        """
+        Called after plugin is loaded to pass the [configuration] section in their plugin info file
+        :param conf: configuration dictionary
+        """
+        self._user = conf.get('user')
+        self._pass = conf.get('pass')
+
+    def create(self, factory_ctx):
+        """
+        Automatically called to create the check function's object
+        :param factory_ctx: (dict) names available for Function instantiation
+        :return: an object that implements a check function
+        """
+        return propartial(AppdynamicsWrapper, username=self._user, password=self._pass)
+
+
+class AppdynamicsWrapper(object):
+    def __init__(self, url, username=None, password=None):
+        self.url = url
+        self.timeout = 3
+
+        self._session = requests.Session()
```

Comment:
this should be hidden, i.e. not accessible from ZMON check command code

make it start with two underscores